components/BaseContentController.php

```diff
@@ -482,4 +488,27 @@ protected function updateLockedComments(ContentActiveRecord $activeRecord, array
         $activeRecord->content->locked_comments = $data['locked_comments'];
         return $activeRecord->content->save();
     }
-}
\ No newline at end of file
+
+    protected function updateScheduledAt(ContentActiveRecord $activeRecord, array $data): bool
+    {
+        if (!isset($data['scheduled_at'])) {
+            return true;
+        }
+
+        $validator = DynamicModel::validateData([
+            'scheduled_at' => $data['scheduled_at']
+        ], [
+            ['scheduled_at', 'datetime', 'format' => 'php:Y-m-d H:i:s']
+        ]);
+
+        if (!$validator->validate()) {
+            $activeRecord->addError('scheduled_at', $validator->getFirstError('scheduled_at'));
+
+            return false;
+        }
+
+        $activeRecord->content->scheduled_at = $data['scheduled_at'];
```

Comment:
@luke- I think we should force `state` to `Content::STATE_SCHEDULED` here.
Also it would be good to implement the updating the column `state` and return it in response.

Reply:
@yurabakhtin Do you think we should make the `content.state` column also updateable via the REST API? 
Since the REST API could potentially be accessible to all users, there might be some issues here.

Reply:
@luke- At least on create a content record we should allow to set the `content.state` as we allow from wall stream form.
Because currently it is strange to set the column `scheduled_at` but `state` is stored as Published.

Reply:
@yurabakhtin Ok